elastic/kibana
# [EPM] Conditionally generate ES index pattern name based on dataset_is_prefix

## Summary

Implements https://github.com/elastic/kibana/issues/88307

This now uses the following index patterns in the generated index templates. A `manifest.yml` file for a data stream `status` looking like this:

```
title: Apache status metrics
release: experimental
type: metrics
[...]
```
results in  `metrics-apache.status-*`, and this:

```
title: Apache status metrics
release: experimental
type: metrics
dataset_is_prefix: true
[...]
```

results in `metrics-apache.status.*-*`

## How to test this

Have a local registry running so that you can edit a data stream `manifest.yml` for testing.

Pick a package that contains a data stream without `dataset_is_prefix` set -- that would be any package we have in the registry now, I'm using the Apache package and the `status` data stream. Install it, and verify that the template for the data stream has been generated correctly:

* in Kibana dev tools, look at the template list with `GET _cat/templates` and fetch a specific template for a data stream, e.g. with `GET _index_template/metrics-apache.status`
* verify that the `index_patterns` property for this template looks like this: `["metrics-apache.status-*"]` and that the priority of the template is `200`.

Now uninstall the package and change the data stream `manifest.yml` to add the `dataset_is_prefix: true` setting. (You might need to restart the local registry after that change.)

* in Kibana dev tools, fetch the template again
* verify that the `index_patterns` property now looks like this: `["metrics-apache.status.*-*"]` and that the priority of the template is `150`.

### Checklist

Delete any items that are not applicable to this PR.
